packages/teams-js/src/public/pages.ts

```diff
@@ -567,9 +671,27 @@ export namespace pages {
      * @param handler - The handler to invoke when the user presses the host client's back button.
      */
     export function registerBackButtonHandler(handler: () => boolean): void {
+      registerBackButtonHandlerHelper(handler, () => {
+        if (!isSupported()) {
+          throw errorNotSupportedOnPlatform;
+        }
+      });
+    }
+
+    /**
+     * @hidden
+     * Undocumented helper function with shared code between deprecated version and current version of the registerBackButtonHandler API.
+     *
+     * @internal
+     * Limited to Microsoft-internal use
+     *
+     * @param handler - The handler to invoke when the user presses the host client's back button.
+     * @param versionSpecificHelper - The helper function containing logic pertaining to a specific version of the API.
+     */
+    export function registerBackButtonHandlerHelper(handler: () => boolean, versionSpecificHelper?: () => void): void {
       ensureInitialized();
-      if (!isSupported()) {
-        throw errorNotSupportedOnPlatform;
+      if (versionSpecificHelper) {
+        versionSpecificHelper();
       }
       backButtonPressHandler = handler;
```

Comment:
I don't think I understand what this handler is for. We're re-assigning it to a new variable, and then ignoring backButtonPressHandler. We then check handler but don't register handler or send it to the host. I don't *think* this is a product of your change, but is it something we need to fix?

Reply:
Hmm the `backButtonPressedHandler` in general seems to follow different conventions. We might want to table this for now and investigate why it follows this format for later.

Reply:
So we do register the handler on line 697. We also assign it to a variable that is used in the `handleBackButtonPress` function. Maybe I'm misunderstanding the concern here?

Reply:
This is very interesting indeed. We should follow up on this separate from this change, since I didn't modify it. I don't think this warrants a fix at this time.

Reply:
@erinha Please open a bug or task to look into this. Fine with leaving it since you haven't added to or changed the problem